modules/packages/DistributedBag.chpl

```diff
@@ -83,183 +94,191 @@
   _____
 
   To use :record:`DistBag`, the initializer must be invoked explicitly to
-  properly initialize the structure. Using the default state without initializing
-  will result in a halt.
+  properly initialize the data structure. By default, one bag instance is
+  initialized per locale, and one segment per task.
 
   .. code-block:: chapel
 
     var bag = new DistBag(int, targetLocales=ourTargetLocales);
 
+  DistBag supports a set of basic methods applying to the segment indexed by
+  ``taskId`` in ``{0..<here.maxTaskPar}``. This guarantees the local DFS ordering.
```

Comment:
OK to use your sentence, but I replaced "accept" by "require", because "accept" sounds like "you can set `taskId` or not" to me, while it is required.

Reply:
If a wrong segment ID is given, we can distinguish two cases:
- `segmentID >= here.maskTaskPar`: this produces `error: halt reached - array index out of bounds`, since segments are internally mapped using the domain `{0..<here.maskTaskPar}`. We could consider supporting user-defined domains, but I think the current implementation is not too restrictive.
- `segmentID < here.maskTaskPar` and `segmentID != taskID`: if the "wrong" calls are done concurrently, this could cause race conditions because the "private" region of each segment is not parallel-safe, and methods like `add` or `remove` act on it. Otherwise, I would say it just work with a different local segment without issue.

Reply:
To be more precise, the only condition to ensure parallel-safety of DistBag (and also DFS) is to map **one segment per task**. It is not mandatory to map segment 0 on task 0, segment 1 on task 1, and so on. Segment 0 on task 1, segment 1 on task 2 and so on is also valid, like any other permutation.

Reply:
I like the words in your last comment. Could you make sure they are in a documentation comment somewhere?